modules/rosapps/applications/sysutils/utils/rosperf/rosperf.c

```diff
@@ -433,15 +431,17 @@ PrintOSVersion(void)
                 return;
               }
 
-            Ret = RegQueryValueExW(hKey, L"ProductType", NULL, NULL,
+            BufLen = sizeof(ProductType);
+            Ret = RegQueryValueExW(hKey, L"ProductType", NULL, &dwType,
                                    (LPBYTE) ProductType, &BufLen);
-            if (ERROR_SUCCESS != Ret || BUFSIZE < BufLen)
+
+            RegCloseKey(hKey);
+
+            if (Ret != ERROR_SUCCESS || dwType != REG_SZ)
```

Comment:
ok you should still somehow check whether BufLen is >= BUFSIZE because the string may have just fit exactly into the buffer and not have been NULL-terminated. And also you need to verify whether it's NULL-terminated anyway because it may not.

Reply:
Thanks for the reminder! I agree no-NULL case is not handled, though that predates this PR.
I'll fix both functions...